nemo/collections/nlp/models/token_classification/ner_model.py

```diff
@@ -0,0 +1,215 @@
+        logits = self(input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask)
+
+        loss = self.loss(logits=logits, labels=labels, loss_mask=loss_mask)
+        tensorboard_logs = {'train_loss': loss, 'lr': self._optimizer.param_groups[0]['lr']}
+        return {'loss': loss, 'log': tensorboard_logs}
+
+    def validation_step(self, batch, batch_idx):
+        """
+        Lightning calls this inside the validation loop with the data from the validation dataloader
+        passed in as `batch`.
+        """
+        input_ids, input_type_ids, input_mask, loss_mask, subtokens_mask, labels = batch
+        logits = self(input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask)
+        val_loss = self.loss(logits=logits, labels=labels, loss_mask=loss_mask)
+
+        subtokens_mask = subtokens_mask > 0.5
+
+        preds = torch.argmax(logits, axis=-1)[subtokens_mask]
+        labels = labels[subtokens_mask]
+        tp, fp, fn = self.class_report(preds, labels)
+
+        tensorboard_logs = {'val_loss': val_loss, 'tp': tp, 'fn': fn, 'fp': fp}
+
+        return {'val_loss': val_loss, 'log': tensorboard_logs}
+
+    def validation_epoch_end(self, outputs):
+        """
+        Called at the end of validation to aggregate outputs.
+        outputs: list of individual outputs of each validation step.
+        """
+        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
+
+        # calculate metrics and log classification report
+        tp = sum(torch.stack([x['log']['tp'] for x in outputs]))
+        fn = sum(torch.stack([x['log']['fn'] for x in outputs]))
+        fp = sum(torch.stack([x['log']['fp'] for x in outputs]))
+        precision, recall, f1 = self.class_report.get_precision_recall_f1(tp, fn, fp, mode='macro')
+
+        tensorboard_logs = {
+            'val_loss': avg_loss,
+            'precision': precision,
+            'f1': f1,
+            'recall': recall,
+        }
+        return {'val_loss': avg_loss, 'log': tensorboard_logs}
+
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config)
+
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config)
+
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        self._test_dl = self.__setup_dataloader(cfg=test_data_config)
+
+    def _setup_dataloader_from_config(self, cfg: DictConfig):
+        text_file = os.path.join(self.data_dir, 'text_' + cfg.prefix + '.txt')
+        label_file = os.path.join(self.data_dir, 'labels_' + cfg.prefix + '.txt')
+
+        if not (os.path.exists(text_file) and os.path.exists(label_file)):
+            raise FileNotFoundError(
+                f'{text_file} or {label_file} not found. The data should be splitted into 2 files: text.txt and \
```

Comment:
Can we have this comments in the example file too to help the user understand the format of the files?

Reply:
That's a good point. This is out of the scope of this PR and this decision should be made across the whole collection since every task requires a specific format described in the docs.